client/post-editor/editor-more-options/copy-post.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSite } from 'state/ui/selectors';
+import AccordionSection from 'components/accordion/section';
+import Button from 'components/button';
+import Dialog from 'components/dialog';
+import FormSectionHeading from 'components/forms/form-section-heading';
+import Gridicon from 'components/gridicon';
+import InfoPopover from 'components/info-popover';
+import PostSelector from 'my-sites/post-selector';
+
+class EditorMoreOptionsCopyPost extends Component {
+
+	static propTypes = {
+		site: PropTypes.object.isRequired,
+		translate: PropTypes.func,
+	};
+
+	constructor() {
+		super( ...arguments );
+		this.state = this.resetState();
+	}
+
+	resetState() {
+		return {
+			selectedPostId: null,
+			showDialog: false,
+		};
+	}
+
+	openDialog = event => {
+		event.preventDefault();
+
+		this.setState( {
+			showDialog: true,
+		} );
+	}
+
+	closeDialog = () => {
```

Comment:
Huh, this is an interesting trick to bind the `this` context. In most React documentation they'll instead recommend that you bind it in the constructor. Any downsides to your approach vs. that recommendation?

```jsx
constructor() {
	super( ...arguments );
	// ...

	this.closeDialog = this.closeDialog.bind( this );
}
```